index.js

```diff
@@ -46,13 +47,22 @@ module.exports = function(plugins, options) {
             };
 
             if (node.tag === 'style' && node.content) {
-                var styles = indentResolve([].concat(node.content).join(''), indent);
-                promise = css.process(styles, options)
-                    .then(function(result) {
-                        node.content = [indentResolve(result.css, indent)];
-                    });
+                var includesStandardType = filterType.test('text/css');
+                var hasTypeAttr = node.attrs && node.attrs.type;
+                var isFilterType = hasTypeAttr && filterType.test(node.attrs.type.trim());
+                var isUndefinedType = !hasTypeAttr || includesStandardType && /^(\s+)?$/.test(node.attrs.type);
+                var isStandardCSS = includesStandardType && (isUndefinedType || isFilterType);
+                var isOther = !isStandardCSS && isFilterType;
 
-                promises.push(promise);
+                if (isStandardCSS || isOther) {
```

Comment:
@michael-ciniawsky your code example would require the empty string checks to be part of the expression, which users likely will not know nor care to cover, but they are necessary for catching `text/css` with spec-compliance.

I can separate the checks by keeping the value null/undefined by default, though.